python/pyspark/cloudpickle/cloudpickle.py

```diff
@@ -1461,7 +1461,7 @@ def dump(obj, file, protocol=None, buffer_callback=None):
     Pickler(file, protocol=protocol, buffer_callback=buffer_callback).dump(obj)
 
 
-def dumps(obj, protocol=None, buffer_callback=None):
+def dumps(obj, protocol=None, buffer_callback=None, dispatch_handlers=None):
```

Comment:
Instead of fixing here, could we just do it within PySpark side? e.g., at `CloudPickleSerializer`

```python
    with io.BytesIO() as file:
        cp = Pickler(file, protocol=protocol, buffer_callback=buffer_callback)

        # Register custom dispatch handlers. This allows per object customization of
        # the serialization.
        if dispatch_handlers is not None:
            for klass, handler in dispatch_handlers.items():
                cp.dispatch_table[klass] = handler

        cp.dump(obj)
        return file.getvalue()
```

Reply:
But then I have to move all of this code over, whats the benefit? It essentially removes `dumps` from cloudpickle.

Reply:
Benefit is less troublesome with moving code around to other repos, e.g., cloudpickle might reject this change.

Reply:
I feel it's a worse change to extract the dumps logic and put it somewhere else.

Reply:
okie no strong opinion if we can safely land them all